sql/src/main/java/io/crate/metadata/Routing.java

```diff
@@ -128,6 +128,42 @@ public boolean containsShards(String nodeId) {
         return false;
     }
 
+    public void updateLocations(Map<String, Map<String, List<Integer>>> existingLocations) {
```

Comment:
not quite sure, but does this not copy the whole locations of a routing into the others?